packages/devextreme/testing/styles/unused-elements.test.ts

```diff
@@ -0,0 +1,158 @@
+import { readFileSync, readdirSync, statSync } from 'fs';
+import { resolve, join, extname } from 'path';
+
+const VAR_NAME_CHARS = 'A-Za-z0-9_-';
+
+const getFilePath = (fileName: string): string => {
+  const relativePath = join(__dirname, '..', '..', fileName);
+  return resolve(relativePath);
+};
+
+const getImagesFromContent = (content: string): string[] => {
+  const dataUriRegex = /data-uri\((?:'(image\/svg\+xml;charset=UTF-8)',\s)?['"]?([^)'"]+)['"]?\)/g;
+  const result: string[] = [];
+  let match = dataUriRegex.exec(content);
+
+  while (match !== null) {
+    const imagePath = getFilePath(match[2]);
+    result.push(imagePath);
+    match = dataUriRegex.exec(content);
+  }
+
+  return result;
+};
+
+const getFilesFromDirectory = (directoryName: string, extensions: string[] = []): string[] => {
+  const fullDirName = join(process.cwd(), directoryName);
+  const result: string[] = [];
+
+  const walkDirectory = (directory: string): void => {
+    readdirSync(directory).forEach((file) => {
+      const absolutePath = join(directory, file);
+      if (statSync(absolutePath).isDirectory()) {
+        walkDirectory(absolutePath);
+      } else if (extensions.length === 0 || extensions.includes(extname(file))) {
+        result.push(absolutePath);
+      }
+    });
+  };
+
+  walkDirectory(fullDirName);
+  return result;
+};
+
+const extractVariables = (filePath: string): string[] => {
+  const content = readFileSync(filePath, 'utf8');
+  const regex = new RegExp(`\\$[${VAR_NAME_CHARS}]*`, 'g');
```

Comment:
may be `\\$[${VAR_NAME_CHARS}]+` more correct pattern? ( * replace  by +  )

Reply:
Good remark, thanks [39d09ae](https://github.com/DevExpress/DevExtreme/pull/26514/commits/39d09ae54f8ce628ffc139ea74d2614109a60835)